broker/network/choria_auth.go

```diff
@@ -508,6 +477,24 @@ func (a *ChoriaAuth) handleProvisioningUserConnectionWithTLS(c server.ClientAuth
 }
 
 func (a *ChoriaAuth) handleProvisioningUserConnection(c server.ClientAuthentication, tlsVerified bool) (bool, error) {
+	if a.provPass == emptyString {
+		return false, fmt.Errorf("provisioning user password not enabled")
+	}
+
+	if a.provisioningAccount == nil {
+		return false, fmt.Errorf("provisioning account is not set")
+	}
+
+	opts := c.GetOpts()
+
+	if opts.Password == emptyString {
+		return false, fmt.Errorf("password required")
+	}
+
+	if a.provPass != opts.Password {
+		return false, fmt.Errorf("invalid provisioner password supplied")
+	}
```

Comment:
some monitor refactoring here to pull out shared code between the 2 handler functions below